userguide/sharing.rst

```diff
@@ -1093,16 +1093,15 @@ settings:
   :menuselection:`Services --> SMB`.
 
 * When the :guilabel:`Browsable to Network Clients` box is checked (the 
-  default), the share will be visible in the server's browse list. If the share 
-  is is flagged as a :guilabel:`home share`, unchecking this box will hide the 
-  [homes] share so that only the dynamically generated home share with the
-  authenticated user's name will be visible (in contrast with [homes]
-  and [<username>] both being visible). 
-
-  The presence of a share in a user's browse list does not grant the user  
-  read or write permissions on the share. Users are still able to directly 
-  access and map shares that are not visible in the browse list by using 
-  the share's UNC path. Hence, this option is of limited security utility.
+  default), the share is visible when the user lists the available shares 
+  through Windows File Explorer or through the *net view* command. If the  
+  share is flagged as a :guilabel:`home share`, unchecking this box hides the 
+  *homes* share so that only the dynamically generated home share with the
+  authenticated username will be visible (in contrast with a share named
+  *homes* and the dynamically-generated home share both being visible). 
```

Comment:
Well, the behavior is confusing as well. ;-) For instance, in my smb.conf I have the share:
[my share]
path = /mnt/dozer/share

It appears in explorer as "My Share". If I check the box "use as a homeshare", the smb4.conf is rewritten so that it has 
[homes]
path = /mnt/dozer/share

After checking the button I have two shares in windows explorer "homes" and "awalker" (my username). If I uncheck "browsable to network clients" in the share configuration, I will only have one share "awalker" because the automatically generated home share inherits the browsable parameter from the [global] section of the smb4.conf.

So the behavior is somewhat complicated and confusing for users who aren't familiar with Samba. [homes] is a special reserved section of the smb.conf file that has this unique behavior. 

So "homes" is a share. "Use as a home share" is a gui element checkbox. "Home share" refers to the share that has been flagged in the FreeNAS UI as a home share.